HAR-4091 supernäkymä

tatut opened the pull request:
Tiekohtainen "supernäkymä" tilannekuvaan.

tatut committed:
Poista turhia requireja

tatut committed:
Palauta tehtävät toteumalle oikein

tatut committed:
Jos sijainti poistuu, poista geometria

Jos TR-kentän :sijainti optiossa oleva atomi häipyy, poista näytetty TR-geometria.

tatut committed:
Lisää tienäkymän karttataso

tatut committed:
Hae tarkastukset tienäkymään

tatut committed:
Refaktoroi valitun asian antamista

Valittu asia annetaan nyt arvon ja tunnisteen sijasta funktiona, jolloin
kartalla-esitettavaan-muotoon kutsuja voi määritellä sen itse.

tatut committed:
Päivitä karttataso kun avataan/suljetaan

tatut committed:
Poista tilannekuva haitarissa välilehden padding

tatut committed:
Kasvata z-index valituille viivoille

tatut committed:
Testaa aukinaisuus mäpeillä indeksien sijasta

tatut committed:
Päivitä jeesql

tatut committed:
Ei indikaattoria historianäkymään

tatut committed:
Anna tapahtuman geometria klikkauskäsittelijälle

tatut committed:
Anna idx kaikille tuloksille

tatut committed:
Avaa tulos kartalta klikattaessa

Kauko committed:
Lisää vanhat-ja-uudet-parametrit mixin

Mixiniä voi käyttää komponentin vanhojen ja uusien parametrien
vertailuun. Tarvittiin alunperin tr-komponenttia varten; komponentin
pitää tehdä uusi reitinhaku, kun sille annettava tr-osoite muuttuu.
Aiemmin haku tehtiin vain on-blurrissa, eli kun kenttää kosketti hiirellä.

Kauko committed:
Tr-kenttä hakee reitin aina, kun tr-osoite muuttuu

Kauko committed:
Uudet-ja-vanhat-parametrit palauttaa parametrit oikeasti vektoreina

tatut committed:
div.inline -> span

tatut committed:
Tarkenna miksi tienäkymä renderöidään frontilla

tatut committed:
pisteet toteutus geometry-collectionille

tatut committed:
TR-osoiteindikaattori karttatyyppi

Kartalla esitettävä muoto TR-osoitteen indikaattorille, joka näyttää
raja-aidat molemmissa päissä

tatut committed:
Lisää TR-osoitteen indikaattori kartalle

tatut committed:
Tue geometry-collection tyylitystä

tatut committed:
Merge branch 'HAR-4091-supernakyma' of github.com:finnishtransportagency/harja into HAR-4091-supernakyma

tatut committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-4091-supernakyma

tatut committed:
Laske kahden pisteen välinen kulma

tatut committed:
Suuntaa aidat tieviivan mukaan

Suuntaa aidat tieviivan mukaisesti (seuraavan pisteen) eikä loppupisteen mukaisesti.

tatut committed:
Korjaa pääl./paik. karttatasojen parametrit

tatut committed:
Siirrä linkkifunktioiden deref oikeaan paikkaan

tatut committed:
Posita logitus

tatut committed:
Korjaa avattujen nostaminen

Nosta avattujen z-index ylemmäs

tatut committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-4091-supernakyma

tatut committed:
Korjaa ajax-loader sijainti

Tee ajax-loaderista inline-block, jotta se näkyy hae napin vieressä eikä
voi mennä pienellä näytöllä content alueen ulkopuolelle.

tatut committed:
Tue eri sijaintia kartan selityksille

Kartan selityksien sijainnin voi asettaa.

tatut committed:
Siirrä selitykset pois infopaneelin tieltä

tatut committed:
Scroll infopaneeliin

tatut committed:
Näytä 2 eri tasoa: valitut ja ei-valitut

Näytä ei-valitut himmeänä (0.4 opacity), valitut värikkäänä.

tatut committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-4091-supernakyma

tatut committed:
Uudenmalliset parametrit kartalle

tatut committed:
Näytä toteuman reittipisteet palloina

tatut committed:
Näytä paremmat tiedot tienäkymän paneelissa

tatut committed:
Ilmoitukset tienäkymään

tatut committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-4091-supernakyma

tatut committed:
Muuta löytymistoleranssia

tatut committed:
Hae turpot tienäkymään

tatut committed:
Tee vetolaatikot-auki ulkoinen atomi

Tätä optiota ei käytetty missään, nyt sillä on mahdollista antaa
ulkoinen atomi vetolaatikoiden aukinaisuudelle.

tatut committed:
Tallenna avatut toteumat ulkoiseen atomiin

Tienäkymästä halutaan linkittää tälle sivulle ja avata valmiiksi

tatut committed:
Poista partial

Tee 2-arity funktio

tatut committed:
Alustava linkkifunktio toteumille

tatut committed:
Nimeä valitse-hallintayksikko paremmin

tatut committed:
Lisää urakka/hallintayksikko toteumiin

tatut committed:
Mahdollista aikavälin ohjelmallinen valinta

tatut committed:
Hae päiväkohtaiset tiedot tieto-ns

Haetaan avattujen toteumien päiväkohtaiset tiedot tietonimiavaruudessa,
fronttikomponentin sijasta.

tatut committed:
Alustava toteumanäkymän avaus

tatut committed:
Toimiva toteuman aukaisu tienäkymästä

tatut committed:
Dokumentoi known issue

tatut committed:
Valitse tpi instanssi koodilla

tatut committed:
Valitse oikea toimenpideinstanssi kun avataan

Kun avataan toteumaa, valitse sen toimenpideinstanssi 3. tason koodin perusteella.

tatut committed:
Palvelu siirtymätietojen hakemiseen

Palvelu, joka palauttaa annetun kokonaishintaisen toteuman tiedot, jotka
tarvitaan siirtymiseen.

tatut committed:
Siirrä siirtymä omaan nimiavaruuteen

Uusi siirtymat ns, jossa tehdään syvänavigaatio. Toteutettu
kokonaishintaiseen toteumaan siirtyminen id:n perusteella.

tatut committed:
Korjaa aikaväli valinnassa

tatut committed:
Poista turhia tietoja, jotka haetaan siirtymässä

tatut committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-4091-supernakyma

tatut committed:
Lisää ei-tuloksia optio infopaneeliin

tatut committed:
Lisää pakollisuus kentille

tatut committed:
Poista ylimääräiset tähdet

Rivitetyssä mallissa tulee required luokan perusteella tähdet. Poistettu ylimääräiset.

tatut committed:
Poista tulokset myös kartalta kun suljetaan

tatut committed:
Kuuntele klikkauksia aina kun tienäkymässä

Estä normaalin infopaneelin tuleminen kun ollaan tienäkymässä, vaikka
tienäkymän tuloksia ei olisikaan haettu.

tatut committed:
Aseta myös aika teksti, jos pvm muuttuu

Jos pvm-aika kentän arvo muuttuu ohjelmallisesti, aseta myös aika-teksti
pvm-tekstin lisäksi.

tatut committed:
Korjaa valittu? ylläpitokohteille

Ylläpitokohteissa käytettiin vielä vanhaa mallia, jossa valittu
annettiin funktiona. Nykyään se on boolean.

tatut committed:
Muuta kartan-kokovalinta defonce määritykseksi

tatut committed:
Kytke kartta kun ollaan näkymässä

Odota siirtymässä, että ollaan oikeasti näkymässä ja kytke kartta
näkyviin vasta sitten. Tämä siksi että toteumien päänäkymä piilottaa kartan.

tatut committed:
Testi atomin odotteluapurille

tatut committed:
Lisää testitiedosto

tatut committed:
Lisää pakota-suunta optiot

tatut committed:
Parempi kartan hiirikuuntelu

Tee hiiren klik/hover kuuntelijoista pino, jolloin vain viimeisimpänä
lisättyä kutsutaan. Muutettu poistamista siten, että käsittelijän
lisääminen palauttaa funktion, jolla juuri lisätty kuuntelija poistetaan.

tatut committed:
Poista muu tilannekuvan kartta kun tienäkymässä

tatut committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-4091-supernakyma

tatut committed:
Korjaa virhe kartan-koko-vaihdettu eventissä

tatut committed:
Tue pakota-suunta myös :pvm-aika kentässä

tatut committed:
Kokeile dockeroida harja-dev ympäristö

Alustava Docker imagen rakennukselle skripti, joka asentaa javan,
postgresin ja viimeisimmän harja tuotantobuildin.

tatut committed:
Skriptit docker testikannalle

Skriptit, jolla docker konttina pyörivän testikannan saa pystytettyä
ja tuhottua. devdb_up luo docker run komennolla uuden kannan, ajaa
siihen flyway migraatiot ja testidatan. devdb_down pysäyttää ja tuhoaa kontin.

Kauko commented on src/cljs/harja/ui/openlayers/featuret.cljs:
```diff
@@ -35,19 +36,26 @@ pienemmällä zindexillä." :const true}
   (doto feature
     (.setStyle (ol.style.Style.
-                #js {:fill (when fill
+                #js {:image (some-> circle circle-style)
```
Mitä tämä tekee?

tatut replied:
Piirtäää ympyrän jos sellain on tyylinä määritelty. Tarkoituksen oli visualisoida reittipisteet palluroina.

Kauko commented on src/cljs/harja/views/kartta/tasot.cljs:
```diff
@@ -212,12 +218,14 @@
       :paikkauskohteet (taso :paikkauskohteet)
       :tr-valitsin (taso :tr-valitsin (inc oletus-zindex))
+      :tienakyma-valitut (taso :tienakyma-valitut)
+      :tienakyma-muut (taso :tienakyma-muut :tienakyma-muut 0.4)
```
Kiinnostaisi kuulla, miksi päädyit tekemään läpinäkyvyyden tasona. Eikö featurekohtainen läpinäkyvyys toiminut?

Musta tuo läpinäkyvyys näytti tienäkymässä todella kivalle, ja se vois toimia muissakin näkymissä hyvin. Toki ei montaa näkymää ole, missä asiat piirretään frontilla, mutta esim laatupoikeamissa se vois toimia hienosti.

tatut replied:
En tiennyt, että featurekohtainen läpinäkyvyys on meillä tuettu. Tuo oli todella helppo jakaa kahteen eri tasoon.

Kauko replied:
Ei se tällä hetkellä olekkaan tuettu, mutta eikös periaatteessa olis voinut esitettäviin asioihin tehdä valittu?-himmeliä muistuttavan avattu?-kokonaisuuden. Jos on avattu, niin otetaan väri alpha-väreistä, eikä puhtaista.

Tuo tasohomma toimii tienäkymässä hyvin, mutta saako sitä kuinka helposti laajennettua muihin näkymiin?

tatut committed:
Dokumentoi docker kehitysympäristöä

tatut committed:
Lisää proxy MML WMTS haulle

Lisää kehitysmoodi komponentti, jossa on proxy MML:n WMTS hauille.
Näin paikallisessa kehityksessä ei enää tarvita nginx:ää ollenkaan,
vaan postgres ja activemq docker imageilla pärjää.

tatut committed:
Lisää odottelua

tatut committed:
Aseta väkisin kun datepickeristä valitaan

Jos pvm-aika kentän arvon on ohjelmallisesti asetettu, silloin aika voi
olla koskematon kun pvm muuttuu eikä datepicker valinta ota vaikutusta.
Lisätty force? parametri asettamiseen, joka asettaa väkisin.

tatut committed:
Palauta TR osoite kok.hint. toteumille

Ota TR osoite mukaan päiväkohtaiselle toteumalistaukselle, jotta se
näkyy lomakkeessa kun toteumaa tarkastellaan.

tatut committed:
Aseta valittu-kokonaishintainen-toteuma nil

tatut committed:
Älä yritä asettaa olemattomia headereita

Kopioi MML responsesta vain ne headerit, jotka siinä on.

tatut committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-4091-supernakyma

tatut committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-4091-supernakyma

tatut committed:
Kuuntele data atomia ja päivitä tekstikentät

Jos pvm-aika komponentin data muuttuu, päivitä tekstikentät vastaamaan
sitä. Tämä siksi jos ohjelmallisesti muutetaan päivämäärää/aikaa, pitää
tekstien muuttua.

tatut committed:
Lisää testejä :pvm-aika tyyppiselle kentälle

tatut committed:
Varmista, että tallennuspalvelu on kutsuttu

tatut committed:
Placeholder testins tienäkymän palvelulle

tatut committed:
Poista valhe README:stä

tatut committed:
Parempi pvm-aika ja testit

Testattu, että pvm-aika kenttä toimii sekä normaalin atomin kanssa että
wrap kanssa oikein. Arvot päivittyvät sekä UI:lta valittaessa että
toiseen suuntaan kun data muuttuu ohjelmallisesti.

tatut committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-4091-supernakyma

tatut committed:
Luo myös testikanta ja template dockeriin

tatut committed:
Yksinkertainen testi tienäkymän palvelulle

tatut committed:
Lisää validointi TR-osoitteelle

tatut committed:
Paranna GEOMETRYCOLLECTION tukea

Aja ST_CollectionHomogenize reitille ennen intersects
tarkistusta. Lisäksi tarkistetaan envelope, koska se on indeksoitu.

tatut committed:
Jos osoite muuttuu, aseta sijainti nil

tatut committed:
Erillinen :ei-haettu arvo sijainnille

Pieni UI-viilaus, virhe näytetään vain oikeasti haetusta arvosta ja
poistetaan heti kun arvo muuttuu.

tatut committed:
Merge branch 'develop' into HAR-4091-supernakyma

Jarzka commented on src/cljs/harja/tiedot/tilannekuva/tienakyma.cljs:
```diff
@@ -1,19 +1,24 @@
 ;; Feature fläg, jolla tienäkymä on pois käytöstä.
 ;; Testailua varten, sen voi kytkeä JS konsolissa päälle.
-(defonce tienakyma-kaytossa? (atom false))
+(defonce tienakyma-kaytossa? (atom true))
```
Poista tämä

Jarzka commented on src/cljs/harja/tiedot/tilannekuva/tienakyma.cljs:
```diff
@@ -31,6 +36,29 @@
+           :valitut-tulokset-kartalla
+           (esitettavat-asiat/kartalla-esitettavaan-muotoon
+            (concat valitut-tulokset
+                    #_(map #(assoc % :tyyppi-kartalla :reittipisteet)
```
Kommentoitu map?

Jarzka commented on src/cljs/harja/tiedot/tilannekuva/tienakyma.cljs:
```diff
@@ -51,8 +79,16 @@
+                       (assoc v :sijainti :ei-haettu)
+                       v)
+
+                     )]
```
Sulut virheellisesti omalla rivillä?

tatut committed:
Poista feature flag

Jarzka commented on src/cljs/harja/tiedot/tilannekuva/tilannekuva_kartalla.cljs:
```diff
@@ -20,6 +20,7 @@
 
 (defonce karttataso-tilannekuva (atom false))
 (defonce haetut-asiat (atom nil))
+(defonce valittu-tila (atom nil))
```
Voisiko tähän kommentoida mahdolliset tilat?

tatut committed:
Poista kommentoitu arvo

tatut committed:
Formatoi

tatut committed:
Dokumentoi kommenttiin eri tilat
